exporters/otlp/otlpmetric/internal/transform/metricdata.go

```diff
@@ -174,17 +174,24 @@ func HistogramDataPoints(dPts []metricdata.HistogramDataPoint) []*mpb.HistogramD
 	out := make([]*mpb.HistogramDataPoint, 0, len(dPts))
 	for _, dPt := range dPts {
 		sum := dPt.Sum
```

Comment:
As I read the PR, I became concerned that there are still allocations happening in this method that #3201 was meant to help with.  I also became concerned that the address of `dPt.Sum` is an alias for a loop-local variable--is that possible i.e., that all data points will be output with the sum-address from the last point in the loop? Is it possible to use a loop index and refer to `&dPts[index].Sum` to avoid allocating?

Reply:
> I also became concerned that the address of dPt.Sum is an alias for a loop-local variable

That is why the address is copied each loop into `sum`. Doing this ensures the correct address is used for each data point, see https://github.com/open-telemetry/opentelemetry-go/pull/3293 for more info.

> Is it possible to use a loop index and refer to `&dPts[index].Sum` to avoid allocating?

Wouldn't `&dPts[index].Sum` also allocate a new reference to the same value?

Reply:
> As I read the PR, I became concerned that there are still allocations happening in this method that #3201 was meant to help with.

I don't think there is a way around allocating to the heap a value here given the OTLP defines the sum, min, and max this way. That is going to be an issue with whatever solution is accepted from #3201.

This approach does prevent a heap allocation in other exporters that do not represent this value as a pointer.